libavcodec/decode.c

```diff
@@ -1395,6 +1442,15 @@ int ff_get_format(AVCodecContext *avctx, const enum AVPixelFormat *fmt)
     memcpy(choices, fmt, (n + 1) * sizeof(*choices));
 
     for (;;) {
+        if (avctx->internal->hwaccel_priv_data &&
+            avctx->hwaccel->caps_internal & HWACCEL_CAP_INTERNAL_ALLOC) {
+            err = hwaccel_realloc_surface(avctx);
+            if (err < 0)
+                av_log(avctx, AV_LOG_WARNING, "Try to re-initialize all.\n");
```

Comment:
change to a better messeage. "on fly format change failed, Try to re-intit...."
But is this really usefful? If we reach here we do not have hw_frames_ctx, who will allocate new one for us?